x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.test.tsx

```diff
@@ -0,0 +1,276 @@
+      mockKea({
+        values: {
+          meta: {
+            page: {
+              current: 6,
+              size: 55,
+              total_pages: 1,
+              total_results: 1004,
+            },
+          },
+        },
+      });
+
+      const wrapper = shallow(<CredentialsList />);
+      const { pagination } = wrapper.find(EuiBasicTable).props();
+      expect(pagination).toEqual({
+        pageIndex: 5,
+        pageSize: 55,
+        totalItemCount: 1004,
+        hidePerPageOptions: true,
+      });
+    });
+
+    it('will default pagination values if `page` is not available', () => {
+      mockKea({
+        values: {
+          meta: {},
+        },
+      });
+
+      const wrapper = shallow(<CredentialsList />);
+      const { pagination } = wrapper.find(EuiBasicTable).props();
+      expect(pagination).toEqual({
+        pageIndex: 0,
+        pageSize: 0,
+        totalItemCount: 0,
+        hidePerPageOptions: true,
+      });
+    });
+  });
+
+  describe('columns', () => {
+    let columns: Array<{
+      render: (token: object) => any;
+      actions: any;
+    }>;
+    const showCredentialsForm = jest.fn();
+    const deleteApiKey = jest.fn();
+
+    beforeEach(() => {
+      mockKea({
+        actions: {
+          showCredentialsForm,
+          deleteApiKey,
+        },
+      });
+      const wrapper = shallow(<CredentialsList />);
+      columns = wrapper.find(EuiBasicTable).props().columns;
+    });
+
+    describe('column 1 (name)', () => {
+      const token = {
+        ...apiToken,
+        name: 'some-name',
+      };
+
+      it('renders correctly', () => {
+        const column = columns[0];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        expect(wrapper.text()).toEqual('some-name');
+      });
+    });
+
+    describe('column 2 (type)', () => {
+      const token = {
+        ...apiToken,
+        type: ApiTokenTypes.Private,
+      };
+
+      it('renders correctly', () => {
+        const column = columns[1];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        expect(wrapper.text()).toEqual('Private API Key');
+      });
+    });
+
+    describe('column 3 (key)', () => {
+      const testToken = {
+        ...apiToken,
+        key: 'abc-123',
+      };
+
+      const subject = (token: object): [any, any] => {
+        const copyMock = jest.fn();
+        const column = columns[2];
+        const wrapper = shallow(<div>{column.render(token)}</div>);
+        const children = wrapper.find(EuiCopy).props().children;
+        const copyEl = shallow(children(copyMock));
+        return [copyMock, copyEl];
+      };
+
+      it('renders the credential and a button to copy it', () => {
+        const [copyMock, copyEl] = subject(testToken);
+        expect(copyEl.find('EuiButtonIcon').props().onClick).toEqual(copyMock);
+        expect(copyEl.text().replace('<EuiButtonIcon />', '')).toBe('abc-123');
```

Comment:
Oh lol I think I see why you had the `<span>` now (to not have to replace `'<EuiButtonIcon />'`). Sorry about that 🤦‍♀️ 

IMO, you can pretty safely simplify the assertion to:

```suggestion
        expect(copyEl.text()).toContain('abc-123');
```

What do you think? (doesn't work for the empy text test below, but I think I'd like us to approach that differently anyway)

Reply:
That's how I had it originally but changed it because of that empty test. But yeah, this approach is better if we just change that branch logic to render nuffin'.

https://github.com/elastic/kibana/pull/79749/commits/a0ff6461eb37e6e3b2418785a5e73c1a8eefcc63?w=1

Reply:
Love it!